src/installer/tests/HostActivation.Tests/DependencyResolution/RidAssetResolution.cs

```diff
@@ -173,17 +289,27 @@ public void MostSpecificRidAssemblySelected(string rid, string expectedPath, boo
         private static string DifferentArchAsset = $"{DifferentArch}/{DifferentArch}Asset.dll";
 
         [Theory]
-        [InlineData(true, true)]
-        [InlineData(true, false)]
-        [InlineData(false, true)]
-        [InlineData(false, false)]
-        public void MostSpecificRidAssemblySelected_ComputedRid(bool includeCurrentArch, bool hasRuntimeFallbacks)
+        [InlineData(true, true, true)]
+        [InlineData(true, true, false)]
+        [InlineData(true, true, null)]
+        [InlineData(true, false, true)]
+        [InlineData(true, false, false)]
+        [InlineData(true, false, null)]
+        [InlineData(false, true, true)]
+        [InlineData(false, true, false)]
+        [InlineData(false, true, null)]
+        [InlineData(false, false, true)]
+        [InlineData(false, false, false)]
+        [InlineData(false, false, null)]
+        public void MostSpecificRidAssemblySelected(bool includeCurrentArch, bool hasRuntimeFallbacks, bool? readRidGraph)
         {
-            // Host relies on the fallback graph to resolve any RID-specific assets that don't exactly match
-            // the current RID, so the OS-only asset remains excluded without the fallback graph currently
+            // When not using the RID graph, the host uses the target OS for which it was built to determine applicable
+            // RIDs that apply, so it can find both the arch-specific and the OS-only assets.
+            // When using the RID graph, the host uses it to resolve any RID-specific assets that don't exactly match
+            // the current RID, so the OS-only asset remains excluded if there are no fallbacks.
             string includedPath = null;
             string excludedPath = $"{CurrentOSAsset};{DifferentArchAsset}";
```

Comment:
I looked at doing this, but it was a bit more complicated than I thought. We'd need to match the fallback RID, since the non-fallback computed RID is ignored if it doesn't exist in the RID graph. But I don't think the tests have a way of knowing the fallback RID right now.

I'll try to do this in a separate change - opened https://github.com/dotnet/runtime/issues/86250